packages/dev/core/src/Misc/sceneSerializer.ts

```diff
@@ -381,6 +397,10 @@ export class SceneSerializer {
      */
     public static SerializeMesh(toSerialize: any /* Mesh || Mesh[] */, withParents: boolean = false, withChildren: boolean = false): any {
         const serializationObject: any = {};
+        serializationObject.meshes = [];
+        serializationObject.transformNodes = [];
+        serializationObject.cameras = [];
+        serializationObject.lights = [];

```

Comment:
Is `SerializeMesh` supposed to also serialize transform nodes, cameras and lights?

It seems we are supposed to pass either a `Mesh` or a `Mesh[]` to the function?

Reply:
We are supposed to pass a Mesh, but the Mesh can have parents and children that aren't meshes, so the serialization would be incorrect for these cases.